blockchain/My-Wallet-V3-iOS
# feat(Homebrew-Shapeshift): IOS-1152 begin adding view controller architecture

## Objective

Added files to be used with services and APIs with architecture similar to `KYCAddressController` and `KYCPersonalDetailsViewController`. Also added `FromToViewDelegateIntermediate` class to use in both the old and new exchange implementations.

## Description

The general approach is to have a view controller `HomebrewExchangeCreateViewController` interact with a coordinator `ExchangeTradeCoordinator` through a `ExchangeTradeDelegate` (from controller to coordinator) and `ExchangeTradeInterface` (from coordinator to controller).

Will add service and API properties to `ExchangeTradeCoordinator` and tests in future PRs (unless reviewers feel that it makes sense to add tests at this point).

## How to Test

Go to exchange (you may have to modify the `if` statement around this if you are in `dev` env https://github.com/blockchain/My-Wallet-V3-iOS/blob/dev/Blockchain/View%20Controllers/ConfirmStateViewController.m#L73) and select various `From` and `To` destinations. You should be able to execute an trade.

## Screenshot

N/A

## Related Information

* Ticket Number: 1152

## Merge Checklist
